products/sentinel_one.py

```diff
@@ -422,9 +428,13 @@ def nested_process_search(self, tag: Tag, criteria: dict, base_query: dict):
                 parameter = PARAMETER_MAPPING[search_field]
                 search_value = all_terms
 
-                if len(terms) > 1:
+                if len(terms) > 1 and not re.findall(r'(?:\" AND)', search_value):
```

Comment:
We can avoid using regex to search for query fields by checking for the parameter keyword at the beginning rather than performing the check as the second step of the if/else

```suggestion
                if parameter == 'query':
                    # Formats queries as (a) OR (b) OR (a) OR (d)
                    search_value = '(' + ') OR ('.join(terms) + ')'
                    operator = 'raw'
                elif len(terms) > 1:
```